### ISSUE TITLE
Unexpected compile error for VB when attribute is surrounded with #If

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#47984]
**Version Used**: 

master

**Steps to Reproduce**:

[SharpLab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBsQDdhJhxBLAMwE8kATEAagB8BJMABwHsEYoACAZSNjjAFgAUIIAKAV2Ao8AYzYBhFNHYBZIgEEYMBHmCj4gtgbY0AdgAt8rNus3bdcQQFFjpOQqhRBggMQ0CbACJwOgDmbAAq5saCADwqAHzeTi6+IuKSMvKKco7OropAA===)

```vb
Imports System

Public Class MyAttribute
    Inherits Attribute
End Class

#If Debug Then
<My>
#End If
Public Class C
End Class
```

**Expected Behavior**:

No compile error

**Actual Behavior**:

    error BC32035: Attribute specifier is not a complete statement. Use a line continuation to apply the attribute to the following statement.
    error BC30012: '#If' block must end with a matching '#End If'.

**Workaround**:

```vb
Imports System

Public Class MyAttribute
    Inherits Attribute
End Class

#If Debug Then
<My>
Public Class C
#Else
Public Class C        
#End If

End Class
```

**Comment by Youssef1313** [comment]
Mono compiler from https://www.onlinegdb.com/online_vb_compiler doesn't complain.

I don't have VS 2008 to test if the native compiler had this bug or not.

**Comment by CyrusNajmabadi** [comment]
i technically think this is by design.   but we'd have to go spelunking.  my understanding is that VB has some constructs that cannot be interrupted (including by the preprocessor).

**Comment by paul1956** [comment]
I looked at the when I implemented _ comments to try to add it and I think it worked in native compiler with _ but Roslyn has never allowed it for VB but does for C#. I reported it as an issue a very long time ago but it never got any attention.

**Comment by sharwell** [comment]
Duplicate of #36365
